src/status_im/contexts/wallet/collectible/view.cljs

```diff
@@ -162,35 +72,40 @@
-        {:keys [collectible-data preview-url
-                collection-data]} collectible
-        {traits           :traits
-         collectible-name :name}  collectible-data
-        {collection-image :image-url
-         collection-name  :name}  collection-data
-        chain-id                  (rf/sub [:wallet/last-collectible-chain-id])]
-    [scroll-page/scroll-page
-     {:navigate-back? true
-      :height         148
-      :page-nav-props {:type        :title-description
-                       :title       collectible-name
-                       :description collection-name
-                       :right-side  [{:icon-name :i/options
-                                      :on-press  #(rf/dispatch
-                                                   [:show-bottom-sheet
-                                                    {:content collectible-actions-sheet
-                                                     :theme   theme}])}]
-                       :picture     preview-url}}
-     [rn/view {:style style/container}
-      [rn/view {:style style/preview-container}
-       [rn/image
-        {:source preview-url
-         :style  style/preview}]]
-      [header collectible-name collection-name collection-image]
-      [cta-buttons]
-      [tabs
-       {:traits   traits
-        :chain-id chain-id}]]]))
+  (let [selected-tab (reagent/atom :overview)]
+    (fn []
+      (let [collectible               (rf/sub [:wallet/last-collectible-details])
+            {:keys [collectible-data preview-url
+                    collection-data]} collectible
+            {collection-image :image-url
+             collection-name  :name}  collection-data
+            {collectible-name :name}  collectible-data]
+        [scroll-page/scroll-page
+         {:navigate-back? true
+          :height         148
+          :page-nav-props {:type        :title-description
+                           :title       collectible-name
+                           :description collection-name
+                           :right-side  [{:icon-name :i/options
+                                          :on-press  #(rf/dispatch
+                                                       [:show-bottom-sheet
+                                                        {:content collectible-actions-sheet
+                                                         :theme   theme}])}]
+                           :picture     preview-url}}
+         [rn/view {:style style/container}
+          [rn/view {:style style/preview-container}
+           [rn/image
+            {:source preview-url
+             :style  style/preview}]]
+          [header collectible-name collection-name collection-image]
+          [cta-buttons]
+          [quo/tabs
+           {:size           32
+            :style          style/tabs
+            :scrollable?    true
+            :default-active @selected-tab
+            :on-change      #(reset! selected-tab %)
```

Comment:
We have to be careful with certain components that we always pass anonymous functions. This one could be extracted outside the renderer, along with the `selected-tab` binding.

If you do so, then Reagent will be able to completely skip processing `quo/tabs`, unless `@selected-tab` really changes. With the current code, `quo/tabs` (which is expensive) will be reprocessed every time `view-internal` re-renders. My suggestion is, if you haven't done yet, double-check this is actually the case, don't trust me, Reagent will skip the rendering completely based on the props equality.

Reply:
DOne